src/lang/builtins_bool.ml

```diff
@@ -27,9 +27,29 @@ let () =
   let register_op name op =
     add_builtin name ~cat:Bool ~descr:"Comparison of comparable values."
       [("", t, None, None); ("", t, None, None)] Lang.bool_t (fun p ->
-        let a = Lang.assoc "" 1 p |> Lang.demeth in
-        let b = Lang.assoc "" 2 p |> Lang.demeth in
-        Lang.bool (op (Lang.compare_values a b)))
+        let a = Lang.assoc "" 1 p in
+        let b = Lang.assoc "" 2 p in
+        let a' = Lang.demeth a in
+        let b' = Lang.demeth b in
+        (* For records, we also compare fields. *)
+        let ans =
+          if a'.Lang.value = Lang.Tuple [] && b'.Lang.value = Lang.Tuple [] then (
+            let r a =
+              let m, _ = Lang_values.V.split_meths a in
+              m
+            in
+            let a = r a in
+            let b = r b in
+            (* TODO: the order is not the expected one on records (for < we want
+               one field to be < and the other to be <=). *)
+            List.for_all
+              (fun (l, v) ->
+                let v' = List.assoc l b in
+                op (Lang.compare_values v v'))
+              a )
```

Comment:
Do you need to also test the other direction, i.e. that all methods from `b` are in `a`?

Reply:
We compare two things of the same type, so `a` and `b` should have the same methods.

Reply:
Ohh okay yes.